core/src/main/java/fr/sncf/osrd/api/PathfindingResult.java

```diff
@@ -0,0 +1,258 @@
+    public final List<RoutePathResult> routePaths = new ArrayList<>();
+    @Json(name = "path_waypoints")
+    public final List<PathWaypointResult> pathWaypoints = new ArrayList<>();
+
+    // make them private final
+    public LineString geographic = null;
+    public LineString schematic = null;
+
+    /**
+     * pathfindingResultMake is used to create a class PathfindingResult with a good format to be sent to the middleware
+     * during this function, we add geometric information in the PathfindingResult
+     * @param finalPathsToGoal contains the routes that are use after the pathfinding
+     * @param infra current infra
+     * @return A pathfindingResult object ready to be sent to the middleware
+     */
+    @SuppressFBWarnings({"BC_UNCONFIRMED_CAST"})
+    public static PathfindingResult pathfindingResultMake(
+            ArrayDeque<FullPathArray<Route, BasicPathNode<Route>>> finalPathsToGoal,
+            Infra infra
+    ) throws InvalidInfraException {
+
+        var res = new PathfindingResult();
+
+        for (var path : finalPathsToGoal) {
+            var routeBeginLoc = pathNodeToRouteLocation(path.pathNodes.get(0));
+            var beginLoc = routeBeginLoc.getTrackSectionLocation();
+            var routeEndLoc = pathNodeToRouteLocation(path.pathNodes.get(path.pathNodes.size() - 1));
+            var endLoc = routeEndLoc.getTrackSectionLocation();
+
+            var routes = new ArrayList<Route>();
+            for (var node : path.pathNodes) {
+                if (routes.isEmpty() || routes.get(routes.size() - 1) != node.edge)
+                    routes.add(node.edge);
+            }
+
+            for (int j = 0; j < routes.size(); j++) {
+                TrackSectionLocation begin = null;
+                TrackSectionLocation end = null;
+                if (j == 0)
+                    begin = beginLoc;
+                if (j == routes.size() - 1)
+                    end = endLoc;
+                var route = routes.get(j);
+                var trackSections = Route.routesToTrackSectionRange(
+                        Collections.singletonList(route), begin, end);
+                if (j == 0) {
+                    // Add the given origin location to the steps output
+                    var firstTrack = trackSections.get(0);
+                    var newStep = new PathWaypointResult(firstTrack.edge,
+                            firstTrack.getBeginPosition());
+                    res.addStep(newStep);
+                }
+                res.add(route, trackSections);
+                if (j == routes.size() - 1) {
+                    // Add the given destination location to the steps output
+                    var lastTrack = trackSections.get(trackSections.size() - 1);
+                    var newStep = new PathWaypointResult(lastTrack.edge,
+                            lastTrack.getEndPosition());
+                    res.addStep(newStep);
+                }
+            }
+        }
```

Comment:
This could be put in a function like:
```java
private static void fillRoutePath(
    ArrayDeque<FullPathArray<Route, BasicPathNode<Route>>> finalPathsToGoal,
    PathfindingResult pathfindingResult
) {
   // Copy paste
}
```